[Security Solution] [EUI Refresh] styled component migration Dashboards

agusruidiazgd opened the pull request:
## Summary

Addresses https://github.com/elastic/kibana/issues/206439
This PR migrates explore page from styled-components to @emotion/styled.
In the process I also convert the kbn/ui-theme json tokens to euiTheme counterparts.

https://github.com/user-attachments/assets/6cdfbf15-9a77-488c-a0b9-42acc3fe1768



<!--ONMERGE {"backportTargets":["8.18","8.x","9.0"]} ONMERGE-->

agusruidiazgd committed:
migrate entity analytics and overview landing page

agusruidiazgd committed:
migrate entity analytics and overview landing page

agusruidiazgd committed:
more changes

agusruidiazgd committed:
migrating overview

agusruidiazgd committed:
migrating overview

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

agusruidiazgd committed:
fix shouldForwardProp

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

jbudz reviewed (approved):
`packages/kbn-babel-preset/styled_components_files.js` LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/271602)
* Commit: a44825778ed11f8c64e41daefc7d5988beb50c88

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/271602#0194b296-2601-45f5-b82b-bc46c9ed57bc)

cc @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271602","number":271602,"commit":"a44825778ed11f8c64e41daefc7d5988beb50c88"}],"number":271602}
buildkite-pr-comment-->

agusruidiazgd committed:
fix types and deprecated props

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/271892)
* Commit: 280937e4ed859472f6abe9c490240da6d6a502ed

### Failed CI Steps
* [Build and Deploy to Cloud](https://buildkite.com/elastic/kibana-pull-request/builds/271892#0194b69a-5e54-4100-98c5-9d00435f5573)

### History
* :broken_heart: [Build #271602](https://buildkite.com/elastic/kibana-pull-request/builds/271602) failed a44825778ed11f8c64e41daefc7d5988beb50c88

cc @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271892","number":271892,"commit":"280937e4ed859472f6abe9c490240da6d6a502ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271602","number":271602,"commit":"a44825778ed11f8c64e41daefc7d5988beb50c88"}],"number":271892}
buildkite-pr-comment-->

agusruidiazgd committed:
Merge branch 'main' into fix/styled-components-migration-206439

agusruidiazgd committed:
update snapshots

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/271958)
* Commit: 2b89c358e29071de94bfb3e9c9391cdc9d34b3fe

### Failed CI Steps
* [Build and Deploy to Cloud](https://buildkite.com/elastic/kibana-pull-request/builds/271958#0194b735-1359-462e-b0e8-0d80a748b539)

### History
* :broken_heart: [Build #271892](https://buildkite.com/elastic/kibana-pull-request/builds/271892) failed 280937e4ed859472f6abe9c490240da6d6a502ed
* :broken_heart: [Build #271602](https://buildkite.com/elastic/kibana-pull-request/builds/271602) failed a44825778ed11f8c64e41daefc7d5988beb50c88

cc @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271958","number":271958,"commit":"2b89c358e29071de94bfb3e9c9391cdc9d34b3fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271892","number":271892,"commit":"280937e4ed859472f6abe9c490240da6d6a502ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271602","number":271602,"commit":"a44825778ed11f8c64e41daefc7d5988beb50c88"}],"number":271958}
buildkite-pr-comment-->

agusruidiazgd commented:
@elasticmachine merge upstream

hop-dev reviewed (approved):
🚀 Thanks for this!

kapral18 committed:
Merge branch 'main' into agusruidiazgd__fix/styled-components-migration-206439

* main: (402 commits)
  [Search]: Fix Number type field to have correct property (#210462)
  Change filter for rule monitoring gaps (#209983)
  Update Logs Explorer deprecation messages (#201307)
  [APM] Remove `error.id` in `getErrorGroupMainStatistics` query as it's not used (#210613)
  [Embeddable] Fix presentation panel styles (#210113)
  [ci] enable Scout reporter for on-merge-unsupported-ftrs (#210627)
  [Fix][Synonyms UI]Add navigation link to the Detail breadcrumb. (#209574)
  chore(dep): bump `store2` from `2.12.0` to `2.14.4` (#210530)
  [scout] adding test helper `@kbn/scout-oblt` package and uptate onboarding tests (#209761)
  [Cloud Security] Asset Inventory table flyout controls  (#208452)
  [ML] Fix model deployment check in file uploader (#209585)
  Updates archive again (#209828)
  [Security Solution] Added concurrency limits and request throttling to prebuilt rule routes (#209551)
  [Search] [Onboarding] Update search api to use EventEmitter instead of Provider (#209784)
  [maps] lazy load map actions (#210252)
  [Cloud Security] Adding telemetry collection condition based on render condition (#208758)
  [Solution nav] Use flyout for Stack Management in Search and Observability solutions (#208632)
  [Search] Fix Add Inference Endpoint API call (#210243)
  [Agentless Connectors] Integration overview panel (#210222)
  [Lens] Restore dynamic colouring by value for Last value agg (#209110)
  ...

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/275350)
* Commit: 7a6ce19b864c2c8cd248fb46dc32ec3689a7a32b
* [Cloud Deployment](https://kibana-pr-208731.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #271958](https://buildkite.com/elastic/kibana-pull-request/builds/271958) failed 2b89c358e29071de94bfb3e9c9391cdc9d34b3fe
* :broken_heart: [Build #271892](https://buildkite.com/elastic/kibana-pull-request/builds/271892) failed 280937e4ed859472f6abe9c490240da6d6a502ed
* :broken_heart: [Build #271602](https://buildkite.com/elastic/kibana-pull-request/builds/271602) failed a44825778ed11f8c64e41daefc7d5988beb50c88

cc @kapral18 @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275350","number":275350,"commit":"7a6ce19b864c2c8cd248fb46dc32ec3689a7a32b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271958","number":271958,"commit":"2b89c358e29071de94bfb3e9c9391cdc9d34b3fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271892","number":271892,"commit":"280937e4ed859472f6abe9c490240da6d6a502ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271602","number":271602,"commit":"a44825778ed11f8c64e41daefc7d5988beb50c88"}],"number":275350}
buildkite-pr-comment-->

kapral18 committed:
chore: update test snapshots and add tests

- Update inline snapshot for `useAnomaliesColumns` test to reflect changes in styled component rendering.
- Extend `matchers` from `@emotion/jest` to make styles visible to toHaveStyleRule matcher
- Extend test coverage for `RiskScore` component by adding a test case to verify background color rendering when `hideBackgroundColor` is false.

kapral18 reviewed (approved):
LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/275538)
* Commit: 49814f25af013176679ca1427d25f3c362c623f0
* [Cloud Deployment](https://kibana-pr-208731.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #275350](https://buildkite.com/elastic/kibana-pull-request/builds/275350) failed 7a6ce19b864c2c8cd248fb46dc32ec3689a7a32b
* :broken_heart: [Build #271958](https://buildkite.com/elastic/kibana-pull-request/builds/271958) failed 2b89c358e29071de94bfb3e9c9391cdc9d34b3fe
* :broken_heart: [Build #271892](https://buildkite.com/elastic/kibana-pull-request/builds/271892) failed 280937e4ed859472f6abe9c490240da6d6a502ed
* :broken_heart: [Build #271602](https://buildkite.com/elastic/kibana-pull-request/builds/271602) failed a44825778ed11f8c64e41daefc7d5988beb50c88

cc @kapral18 @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275538","number":275538,"commit":"49814f25af013176679ca1427d25f3c362c623f0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275350","number":275350,"commit":"7a6ce19b864c2c8cd248fb46dc32ec3689a7a32b"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271958","number":271958,"commit":"2b89c358e29071de94bfb3e9c9391cdc9d34b3fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271892","number":271892,"commit":"280937e4ed859472f6abe9c490240da6d6a502ed"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/271602","number":271602,"commit":"a44825778ed11f8c64e41daefc7d5988beb50c88"}],"number":275538}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.18, 8.x, 9.0

https://github.com/elastic/kibana/actions/runs/13288411997

kapral18 commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/210886">](https://github.com/elastic/kibana/pull/210886)|
|❌|8.x|Conflict resolution was aborted by the user|
|❌|8.18|Conflict resolution was aborted by the user|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 208731
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->

kapral18 commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/210967">](https://github.com/elastic/kibana/pull/210967)|
|❌|8.18|An unhandled error occurred. Please see the logs for details|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 208731
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->

kapral18 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.18|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/210983">](https://github.com/elastic/kibana/pull/210983)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.4 -->
